### ISSUE TITLE
ZeroNet is not "portable"

### ISSUE THREAD

**Issue by ghost** [HelloZeroNet/ZeroNet#2515]
While ZeroNet calling for backing up users.json, that file will not help you anything! 
If you clear the data folder and than restore your backup you will need to open each and all sites individually to see them in UI. 

Moving ZeroNet between computers is just as impossible, you should not move users.json to multiple computers and so **ZeroNet is not portable**. 

Even if you able to restore your data folder, to add your sites to another node will again require you to open each site, one-by-one and click on the confirmation button to add the site. 

ZeroNet is NOT serving the data folder until you don't open each of your site in UI, which is a disaster.

**Comment by SuperSandro2000** [comment]
> that file will not help you anything!

For me it has a lot of sites listed which I could just visit in the UI and re-add them. Sure better than losing access to your account.

> Moving ZeroNet between computers is just as impossible

As what? Also if you move the entire directory it is very portable and you can just open in on the next machine.

> Even if you able to restore your data folder, to add your sites to another node will again require you to open each site, one-by-one and click on the confirmation button to add the site.

Which is only inconvenient and you don't lose any data which is the purpose of a backup.

Sure some convenience around this could be added to make it easier for the end user.

**Comment by ghost** [comment]
@SuperSandro2000 you should **never copy your private keys** with users.json to any other computer! 

ZeroNet is not portable for the reason I mentioned. If you not copy the entire folder (including private keys) than you will need to manually add each site in UI and download any additional files. 
So no, is not portable especially if you have hundreds of sites! **Will take hours just to open and download them all!**

**Comment by ghost** [comment]
yes @filips123? :thinking:

**Comment by filips123** [comment]
> you should never copy your private keys with users.json to any other computer!

Why?

> If you not copy the entire folder (including private keys) than you will need to manually add each site in UI and download any additional files.

So then you copy entire folder?

**Comment by ghost** [comment]
@filips123 because **you should never copy private keys** like how you copy other files! Only one node is enough for signing, while others should just fetch the fresh content. The problem is that a newly installed ZeroNet node will require you to open each and all sites and confirm that you wish to add them to that node.

@imachug there should be a command to **bulk** add sites to a node!

**Comment by filips123** [comment]
> because you should never copy private keys like how you copy other files!

Because why?

> there should be a command to bulk add sites to a node!

Or you can just make simple Python or Bash script which loops through sites in list and creates HTTP request to ZeroNet to load it.

**Comment by ghost** [comment]
@filips123 
> Or you can just make simple Python or Bash script which loops through sites in list and creates HTTP request to ZeroNet to load it.

Is not that simple! You need to click a button manually to add a site to your node!

**Comment by filips123** [comment]
No, HTTP request to `http://127.0.0.1:43110/site-address` is enough.

**Comment by ghost** [comment]
Is not enough @filips123 :confused: please check it out! git clone to somewhere and add your site, you will see that need confirmation.

**Comment by Merith-TK** [comment]
Well.. i wouldve loved to comment but its a ghost user... meaning he probably was trolling
